### PR TITLE
Code cleanup as a preparation for single node upgrade

### PR DESCRIPTION
**Pull request by jsuchome**

### CODE REVIEW

**Comment by jsuchome** [comment]
[(5) Allow user to choose which compute nodes to upgrade](https://trello.com/c/IZ8W5tQz/224-5-allow-user-to-choose-which-compute-nodes-to-upgrade)

File: crowbar_framework/spec/models/api/upgrade_spec.rb [hunk 1]
```diff
@@ -519,6 +521,29 @@
...
       allow_any_instance_of(Node).to receive(:upgraded?).and_return(true)
+      allow(Api::Upgrade).to receive(:prepare_all_compute_nodes).and_return(true)
+      allow(Api::Upgrade).to receive(:upgrade_all_compute_nodes).and_return(true)
+      allow(Api::Upgrade).to receive(:finalize_nodes_upgrade).and_return(true)
+      allow_any_instance_of(Crowbar::UpgradeStatus).to receive(:end_step).and_return(true)
+
+      expect(subject.class.nodes_without_delay).to be true
+    end
+
+    it "during preparation of compute nodes upgrade , detect that they are upgraded" do
```
Comment (MaximilianMeister) [inline crowbar_framework/spec/models/api/upgrade_spec.rb]: i'd rephrase this a bit like:
```ruby
it "detects that the nodes are already upgraded during preparation" do
```

or make the preparation a context like:
```ruby
context "during preparation of compute nodes upgrade" do
  it "detects that the nodes are already upgraded" do
```

...but having just one example in a context is also not preferable, but it sounds much better at least

Comment (jsuchome) [inline crowbar_framework/spec/models/api/upgrade_spec.rb]: I'll go for the first one, there's already one context for upgrading nodes

File: crowbar_framework/app/models/api/upgrade.rb [hunk 2]
```diff
@@ -823,14 +824,58 @@ def evacuate_network_node(controller, network_node, delete_namespaces = false)
...
 
+      def prepare_all_compute_nodes
+        ["kvm", "xen"].each do |virt|
+          prepare_compute_nodes virt
+        end
+      end
+
+      # Prepare the compute nodes for upgrade by upgrading necessary packages
+      def prepare_compute_nodes(virt)
+        save_upgrade_state("Preparing compute nodes of #{virt} type for upgrade")
```
Comment (rhafer) [inline crowbar_framework/app/models/api/upgrade.rb]: I'd suggest ```"Preparing #{virt} compute nodes for upgrade"``` or  ```"Preparing compute nodes of type: #{virt} for upgrade"```. Though you might want to check back with a native speaker.

Similar it true for all the other place using ```"#{virt} type"```

File: crowbar_framework/app/models/api/upgrade.rb [hunk 3]
```diff
@@ -823,14 +824,58 @@ def evacuate_network_node(controller, network_node, delete_namespaces = false)
...
+        # remove upgraded compute nodes
+        compute_nodes.select! { |n| !n.upgraded? }
+        if compute_nodes.empty?
+          save_upgrade_state(
+            "All compute nodes of #{virt} type are already upgraded."
+          )
+          return
+        end
+
+        # First batch of actions can be executed in parallel for all compute nodes
```
Comment (rhafer) [inline crowbar_framework/app/models/api/upgrade.rb]: s/First/This

File: crowbar_framework/app/models/api/upgrade.rb [hunk 4]
```diff
@@ -823,14 +824,58 @@ def evacuate_network_node(controller, network_node, delete_namespaces = false)
...
+            "/usr/sbin/crowbar-prepare-repositories.sh",
+            120
+          )
+          save_upgrade_state("Repositories prepared successfully.")
+          execute_scripts_and_wait_for_finish(
+            compute_nodes,
+            "/usr/sbin/crowbar-pre-upgrade.sh",
+            300
+          )
+          save_upgrade_state("Services at compute nodes upgraded and prepared.")
```
Comment (rhafer) [inline crowbar_framework/app/models/api/upgrade.rb]: s/at/on

File: crowbar_framework/app/models/api/upgrade.rb [hunk 5]
```diff
@@ -823,14 +824,58 @@ def evacuate_network_node(controller, network_node, delete_namespaces = false)
...
+        rescue StandardError => e
+          raise_node_upgrade_error(
+            "Error while preparing services at compute nodes. " + e.message
```
Comment (rhafer) [inline crowbar_framework/app/models/api/upgrade.rb]: s/at/on

File: crowbar_framework/app/models/api/upgrade.rb [hunk 6]
```diff
@@ -864,53 +909,41 @@ def upgrade_compute_nodes(virt)
...
-            300
-          )
-          save_upgrade_state("Services at compute nodes upgraded and prepared.")
-        rescue StandardError => e
-          raise_node_upgrade_error(
-            "Error while preparing services at compute nodes. " + e.message
-          )
-        end
-
         # Next part must be done sequentially, only one compute node can be upgraded at a time
```
Comment (rhafer) [inline crowbar_framework/app/models/api/upgrade.rb]: s/Next/This

File: crowbar_framework/app/models/api/upgrade.rb (hunk 2 above)
Comment (jsuchome) [inline crowbar_framework/app/models/api/upgrade.rb]: I'll change it, but it's just log line anyway...
